litr/src/main/java/com/linkedin/android/litr/render/PassthroughSoftwareRenderer.java

```diff
@@ -49,34 +51,58 @@ public void renderFrame(@Nullable Frame frame, long presentationTimeNs) {
             Log.e(TAG, "Null or empty input frame provided");
             return;
         }
-        int tag = encoder.dequeueInputFrame(FRAME_WAIT_TIMEOUT);
-        if (tag >= 0) {
-            Frame outputFrame = encoder.getInputFrame(tag);
-            if (outputFrame == null) {
-                Log.e(TAG, "No input frame returned by an encoder, dropping a frame");
-                return;
-            }
 
-            ByteBuffer inputBuffer = frame.buffer.asReadOnlyBuffer();
-            inputBuffer.rewind();
-            outputFrame.buffer.put(inputBuffer);
-
-            outputFrame.bufferInfo.set(
-                    0,
-                    frame.bufferInfo.size,
-                    TimeUnit.NANOSECONDS.toMicros(presentationTimeNs),
-                    frame.bufferInfo.flags);
-            encoder.queueInputFrame(outputFrame);
-        } else {
-            switch (tag) {
-                case MediaCodec.INFO_TRY_AGAIN_LATER:
-                    Log.e(TAG, "Encoder input frame timeout, dropping a frame");
-                    break;
-                default:
-                    Log.e(TAG, "Unhandled value " + tag + " when receiving decoded input frame");
-                    break;
+        ByteBuffer inputBuffer = null;
+        int capacity;
+        boolean isNewInputFrame = true;
+        boolean areBytesRemaining;
+        do {
+            areBytesRemaining = false;
+            int tag = encoder.dequeueInputFrame(FRAME_WAIT_TIMEOUT);
+            if (tag >= 0) {
+                Frame outputFrame = encoder.getInputFrame(tag);
+                if (outputFrame == null) {
+                    Log.e(TAG, "No input frame returned by an encoder, dropping a frame");
+                    return;
+                }
+
+                if (isNewInputFrame) {
+                    isNewInputFrame = false;
+                    inputBuffer = frame.buffer.asReadOnlyBuffer();
+                    inputBuffer.rewind();
+                }
+
+                if (outputFrame.buffer.remaining() < inputBuffer.remaining()) {
+                    capacity = outputFrame.buffer.remaining();
+                    inputByteBuffer = new byte[capacity];
+                    inputBuffer.get(inputByteBuffer, 0, capacity);
+
+                    outputFrame.buffer.put(inputByteBuffer);
```

Comment:
here, you will have to use `outputFrame.buffer.put(inputByteBuffer, 0, capacity);`